### PR TITLE
[Cherry-pick 1.6]Fix op run log when memory optimization strategy is enabled

### PR DESCRIPTION
**Pull request by sneaxiy**
Cherry-pick #20695.

### CODE REVIEW

File: paddle/fluid/operators/op_debug_string_test.cc [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  auto before_run_str = op->DebugStringEx(&scope);
+  LOG(INFO) << before_run_str;
+  ASSERT_TRUE(before_run_str.find(unknown_dtype) != std::string::npos);
+
+  op->Run(scope, place);
+  platform::DeviceContextPool::Instance().Get(place)->Wait();
+
+  auto after_run_str = op->DebugStringEx(&scope);
+  LOG(INFO) << after_run_str;
+  ASSERT_TRUE(after_run_str.find(unknown_dtype) != std::string::npos);
```
Comment (gongweibao) [inline paddle/fluid/operators/op_debug_string_test.cc:67]: No test for `GetDtype(*scope, var_name)`

Comment (sneaxiy) [reply paddle/fluid/operators/op_debug_string_test.cc:67]: image

Comment (sneaxiy) [reply paddle/fluid/operators/op_debug_string_test.cc:67]: I would add more `ASSERT_TRUE` later. The output contains the case `GetDtype(*scope, var_name)`.

**Review by gongweibao — APPROVED** [review]
LGTM

**Pull request merged**